server/fleet/apple_mdm.go

```diff
@@ -278,9 +330,29 @@ func (cp MDMAppleConfigProfile) AuthzType() string {
 	return "mdm_apple_config_profile"
 }
 
-func (cp MDMAppleConfigProfile) Validate() error {
-	// TODO(sarah): Additional validations for PayloadContent (e.g., screening out FileVault payloads)
-	// should be handled here
+// ScreenPayloadTypes screens the profile's PayloadContent and returns an error if it
+// detects certain PayloadTypes related to FileVault settings.
+func (cp MDMAppleConfigProfile) ScreenPayloadTypes() error {
+	pct, err := cp.Mobileconfig.GetPayloadTypes()
+	if err != nil {
+		switch {
+		case strings.Contains(err.Error(), "empty PayloadContent") || strings.Contains(err.Error(), "encrypted PayloadContent"):
```

Comment:
we should be returning specific errors and using `errors.Is` instead of matching the error message

```go
if !errors.Is(err, ErrEmptyPayloadContent) || !errors.Is(err, ErrEncryptedPayloadContent) {
    return err
}
```